benchmark/clblast/blas3/trsm.cpp

```diff
@@ -0,0 +1,190 @@
+/**************************************************************************
+ *
+ *  @license
+ *  Copyright (C) 2021 Codeplay Software Limited
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  For your convenience, a copy of the License has been included in this
+ *  repository.
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  SYCL-BLAS: BLAS implementation using SYCL
+ *
+ *  @filename trsm.cpp
+ *
+ **************************************************************************/
+
+#include "../utils.hpp"
+
+template <typename scalar_t>
+std::string get_name(char side, char triangle, char transpose, char diagonal,
+                     index_t m, index_t n) {
+  std::ostringstream str{};
+  str << "BM_Trsm<" << blas_benchmark::utils::get_type_name<scalar_t>() << ">/"
+      << side << "/" << triangle << "/" << transpose << "/" << diagonal << "/"
+      << m << "/" << n;
+  return str.str();
+}
+
+template <typename scalar_t>
+void run(benchmark::State& state, ExecutorType* executorPtr, char side,
+         char triangle, char transpose, char diagonal, index_t m, index_t n,
+         scalar_t alpha, bool* success) {
+  // Standard test setup.
+  index_t lda = side == 'l' ? m : n;
+  index_t ldb = m;
+  index_t k = side == 'l' ? m : n;
+
+  using data_t = utils::data_storage_t<scalar_t>;
+
+  const int sizeA = k * lda;
+  const int sizeB = n * ldb;
+
+  // Matrices
+  std::vector<data_t> a(sizeA);
+  std::vector<data_t> b = blas_benchmark::utils::random_data<data_t>(sizeB);
+
+  const scalar_t diagValue =
+      diagonal == 'u' ? data_t{1}
+                      : blas_benchmark::utils::random_scalar<scalar_t>(
+                            scalar_t{1}, scalar_t{10});
+
+  blas_benchmark::utils::fill_trsm_matrix(a, k, lda, triangle, diagValue,
+                                          scalar_t{0});
```

Comment:
Same as the clblas benchmark these should maybe be ```data_t``` though that might not be all that's needed for half to work with these.

Reply:
Done